tests/unit/database_tests.py

```diff
@@ -202,6 +202,11 @@ def test_create_db_delete_db(self):
             # No issue should arise if attempting to create existing database
             db_2 = db.create()
             self.assertEqual(db, db_2)
+            # If we use throw_on_exists=True, it will arise a
+            # CloudantDatabaseException if the database already exists.
+            with self.assertRaises(CloudantDatabaseException) as cm:
+                db.create(throw_on_exists=True)
+            self.assertEqual(cm.exception.status_code, 100)
```

Comment:
I've opened #287 since the exception is 412 when a database already exists.  This exception happens [here](https://github.com/cloudant/python-cloudant/blob/master/src/cloudant/database.py#L343) and we should be preserving this status code when raised by `CloudantDatabaseException` and `CloudantClientException`.